qualang_tools/loops/loops.py

```diff
@@ -90,12 +90,17 @@ def from_array(var, array):
                 "When using logarithmic increments with QUA integers, the resulting values will slightly differ from the ones in numpy.logspace() because of rounding errors. \n Please use the get_equivalent_log_array() function to get the exact values taken by the QUA variable and note that the number of points may also change."
             )
             if step > 1:
-                return (
-                    var,
-                    round(start),
-                    var < round(stop) * np.sqrt(float(step)),
-                    Cast.mul_int_by_fixed(var, float(step)),
-                )
+                if int(round(start) * float(step)) == int(round(start)):

```

Comment:
Should this be a mutiplication or addition?

In my head, you should check to see if the ith variable and i+1th value are the same, i.e.
`if int(round(start) + float(step)) == int(round(start)):`.

Also, this should be checked not just for the 0th and 1st, but all successive values?

Maybe I'm confused because logspace and linspace start/step represent different things...

Reply:
With log scan the syntax is `a[i+1] = step * a[i]` so here I just want to check that the step is large enough so that two successive value are not equal, otherwise the program will hang (a[i+1] = a[i]). When step>1, checking the first iteration is enough because the difference between two successive values will increase

Reply:
Oh yep I see that now. All good.